src/Microsoft.ComponentDetection.Detectors/pnpm/PnpmParsingUtilities.cs

```diff
@@ -50,7 +69,65 @@ private static (string Name, string Version) ExtractNameAndVersionFromPnpmPackag
         return (normalizedPackageName, packageVersion);
     }
 
-    private static (string PackageVersion, int VersionIndex) GetPackageVersion(string[] pnpmComponentDefSections)
+    /// <summary>
+    /// Parse a pnpm dependency path.
+    /// </summary>
+    /// <param name="pnpmDependencyPath">A pnpm dependency path of the form "/@optional-scope/package-name@version(optional-ignored-data)(optional-ignored-data)".</param>
+    /// <returns>Data parsed from path.</returns>
+    public static DetectedComponent CreateDetectedComponentFromPnpmPathV6(string pnpmDependencyPath)
+    {
+        /*
+         * The format is documented at https://github.com/pnpm/spec/blob/master/dependency-path.md.
+         * At the writing it does not seem to reflect changes which were made in lock file format v6:
+         * See https://github.com/pnpm/spec/issues/5.
+         */
+
+        // Strip parenthesized suffices from package. These hold peed dep related information that is unneeded here.
+        // An example of a dependency path with these: /webpack-cli@4.10.0(webpack-bundle-analyzer@4.10.1)(webpack-dev-server@4.6.0)(webpack@5.89.0)
+        var fullPackageNameAndVersion = pnpmDependencyPath.Split("(")[0];
```

Comment:
is this guaranteed to always have [0]? do we need to do index checks beforehand?

Reply:
My understanding is splitting a string always returns at least one part, even if there is no separator or the string is empty, so this should be fine. There likely is a more explicit way to do this (for example with a regular expression), but this should work fine.